src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -184,6 +170,48 @@ public int? MaximumIterationsPerRequest
         }
     }
 
+    /// <summary>
+    /// Gets or sets the maximum number of consecutive iterations that are allowed to fail with an error.
+    /// </summary>
+    /// <value>
+    /// The maximum number of consecutive iterations that are allowed to fail with an error.
+    /// The default value is 3.
+    /// </value>
+    /// <remarks>
+    /// <para>
+    /// When function invocations fail with an exception, the <see cref="FunctionInvokingChatClient"/>
+    /// continues to make requests to the inner client, optionally supplying exception information (as
+    /// controlled by <see cref="IncludeDetailedErrors"/>). This allows the <see cref="IChatClient"/> to
+    /// recover from errors by trying other function parameters that may succeed.
+    /// </para>
+    /// <para>
+    /// However, in case function invocations continue to produce exceptions, this property can be used to
+    /// limit the number of consecutive failing attempts. When the limit is reached, the exception will be
+    /// rethrown to the caller.
+    /// </para>
+    /// <para>
+    /// If the value is set to zero, all function calling exceptions immediately terminate the function
+    /// invocation loop and the exception will be rethrown to the caller.
```

Comment:
"and the exception will be rethrown to the caller"

Do you want to say anything about there possibly being multiple exceptions and them all being wrapped in an aggregate?

Also, did you consider not using an aggregate when only a single exception is incurred? I know we always use an aggregate with Task.Wait and friends, but I'm not sure it was the right choice.

Also seems like when AllowConcurrentInvocation is false (the default), we might just want to let the exception propagate if the limit has been reached?

Reply:
If a single exception propagates, mostly just implementation detail, albeit with an extra catch/throw in the mix and the resulting impact on diagnostics. Not a big deal, though.

Reply:
I guess it would make more of a behavioral difference because if there are multiple function calls in the current iteration, propagating the first exception directly would stop subsequent ones from running at all.

This is *probably* desirable though it's quite a subtle distinction. I've implemented it.